File: src/main/java/shmax/food/NanoFoodPiece.java
package shmax.food;

import shmax.controllers.Main;
import javafx.geometry.Point2D;
import javafx.scene.layout.Pane;
import javafx.scene.paint.Color;
import javafx.scene.shape.Rectangle;

public class NanoFoodPiece extends Pane {

    private boolean eaten = false;
    private Point2D position;
    private float foodValue = (float) Math.random()+.5f;

    public NanoFoodPiece() {
        Rectangle graphics = new Rectangle(11, 11, Color.rgb(150, 144, 107));
        double scale = foodValue;
        graphics.setScaleX(scale);
        graphics.setScaleY(scale);
        getChildren().add(graphics);
        setTranslateX(Math.random()*1200-600+(Main.BTarget!=null? Main.BTarget.getTranslateX(): Main.MCPTarget!=null? Main.MCPTarget.getTranslateX():0));
        setTranslateY(Math.random()*1200-600);
        position = new Point2D(getTranslateX()+graphics.getWidth(),getTranslateY()+graphics.getHeight());
    }

    public double getFoodValue() {
        return foodValue;
    }
    public Point2D getPosition() {
        return position;
    }
    public boolean isEaten() {
        return eaten;
    }

    public void destroy(){
        eaten = true;
    }
}

File: src/main/kotlin/shmax/food/NanoFoodPiece.kt
package shmax.food

import javafx.geometry.Point2D
import javafx.scene.layout.Pane
import javafx.scene.paint.Color
import javafx.scene.shape.Rectangle
import shmax.controllers.MainLoop

class NanoFoodPiece : Pane() {
    var eaten: Boolean = false
    val position: Point2D
    val foodValue: Double = Math.random() + .5

    init {
        val graphics = Rectangle(11.0, 11.0, Color.rgb(150, 144, 107))
        graphics.scaleX = foodValue
        graphics.scaleY = foodValue

        children.add(graphics)

        val spawnTargetX = MainLoop.bacteriaTarget?.translateX ?: MainLoop.multicellularTarget?.translateX ?: 0.0
        val spawnTargetY = MainLoop.bacteriaTarget?.translateY ?: MainLoop.multicellularTarget?.translateY ?: 0.0

        translateX = spawnTargetX + (Math.random() * 1200 - 600)
        translateY = spawnTargetY + (Math.random() * 1200 - 600)

        position = Point2D(translateX + graphics.width / 2, translateY + graphics.height / 2)
    }
}
